### ISSUE TITLE
helm-find-files fails when at @uref in Emacs 25.1

### ISSUE THREAD

**Issue by fice-t** [emacs-helm/helm#1638]
## Expected behavior
`helm-find-files` is called and candidates are shown in a helm buffer.

## Actual behavior from emacs-helm.sh if possible
Error: `tramp-error: Not a Tramp file name: "/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}"`

## Steps to reproduce (recipe)
./emacs-helm.sh (with Emacs 25.1+)
Call `helm-find-files` with point in `@uref` in `@uref{http://www.doc.ic.ac.uk/lab/cplus/c++.rules/}`

## Backtraces if some (M-x toggle-debug-on-error)
```elisp
Debugger entered--Lisp error: (user-error "Not a Tramp file name: \"/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}\"")
  signal(user-error ("Not a Tramp file name: \"/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}\""))
  tramp-error(nil user-error "Not a Tramp file name: \"%s\"" "/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}")
  apply(tramp-error nil user-error "Not a Tramp file name: \"%s\"" "/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}")
  tramp-dissect-file-name("/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}" t)
  tramp-find-foreign-file-name-handler("/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}")
  tramp-file-name-handler(file-remote-p "/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}" nil nil)
  file-remote-p("/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}")
  helm-find-files-input("/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}" #("uref{http://www.doc.ic.ac.uk/lab/cplus/c" 0 40 (fontified t)))
  helm-find-files-initial-input()
  helm-find-files(nil)
  funcall-interactively(helm-find-files nil)
  call-interactively(helm-find-files nil nil)
  command-execute(helm-find-files)
```

## Describe versions of helm, emacs, operating system etc.
Latest version on MELPA on Emacs 25.1

**Comment by thierryvolpiatto** [comment]
fice-t <notifications@github.com> writes:

> ## Actual behavior from emacs-helm.sh if possible
> Error: `tramp-error: Not a Tramp file name: "/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}"`

It looks it is a tramp problem, `file-remote-p` send a user error in such
case where it shoud return nil.
I could add a rule at the helm level but the word uref is not involved:

    (file-remote-p "/something{http:/www.doc.ic.ac.uk/}")

produce the same error.
So I think it would be better if fixed at emacs upstream level.
It is reproductible as well on 24.1.

-- 
Thierry

**Comment by thierryvolpiatto** [comment]
fice-t <notifications@github.com> writes:

> ## Steps to reproduce (recipe)
> ./emacs-helm.sh (with Emacs 25.1+)
> Call `helm-find-files` with point in `@uref` in `@uref{http://www.doc.ic.ac.uk/lab/cplus/c++.rules/}`

My bad, it is not reproductible with emacs-24.5 inside this:

`@uref{http://www.doc.ic.ac.uk/lab/cplus/c++.rules/}`

but only inside this:

`/uref{http:/www.doc.ic.ac.uk/lab/cplus/c++.rules/}`

and on emacs-25.1 it is reproductible on both.

So it looks, for what you reported `@foo{http://something}` it is an
emacs-25 bug only and with `/foo{http://something}` (which should return
nil as well) it a bug for both versions 24.5 and 25.1.

-- 
Thierry

**Comment by thierryvolpiatto** [comment]
The problem is coming from `tramp-file-name-structure` regexp which
doesn't match `@foo{http://www.bar.com}` in `tramp-dissect-file-name`:

    (let ((match (string-match (nth 0 tramp-file-name-structure) name)))
      (unless match (tramp-user-error nil "Not a Tramp file name: \"%s\"" name))

-- 
Thierry

**Comment by fice-t** [comment]
I submitted a [bug](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=24889) in Emacs and it was fixed in master (26.1).
